### PR TITLE
rustc: Fix an ICE when -o bare-path

### PR DESCRIPTION
**Pull request by richo**
rustc will ICE if you specify an outfile path that is bare without a
directory. As a workaround, before this -o ./foo will work

It wasn't clear to me where I could put a test that actually invokes rustc from a shell, although I think I can add doctests to that machinery in librustc_driver that will arrange for this to be called with arguments that would trigger the ICE

### CODE REVIEW

**Comment by pnkfelix** [comment]
@richo you can put tests that invoke `rustc` from a shell into `src/tests/run-make`; those tests are all makefile driven.

**Comment by richo** [comment]
Thanks @pnkfelix I added the test

**Comment by pnkfelix** [comment]
@richo our `make tidy` script is pretty unforgiving; you need to add a license to `src/run-make/bare-outfile/foo.rs`

**Comment by pnkfelix** [comment]
(p.s. even though `make check` can take a long time, `make tidy` is fast enough that I recommend running it locally before you push, if you remember.)

**Comment by richo** [comment]
Ooof, my bad! Thanks for the ping

File: src/librustc_driver/driver.rs [hunk 1]
```diff
@@ -954,8 +954,11 @@ pub fn build_output_filenames(input: &Input,
             if *odir != None {
                 sess.warn("ignoring --out-dir flag due to -o flag.");
             }
+
+            let cur_dir = Path::new(".");
```
Comment (alexcrichton) [inline src/librustc_driver/driver.rs]: This default is actually more appropriately the empty string due to the way that normalization works currently.

**Comment by alexcrichton** [comment]
@pnkfelix yes thanks for the r?, this is a known issue with the path module and I discussed it at great length with @aturon. This is considered a bug of the module and @aturon is working on a fix for it soon (which will cause this to "just work").

I ran into this ICE a few days ago and decided to just let it be while the path module is fixed, but I'm also OK landing this in the interim.

**Comment by richo** [comment]
I'd rather land this (or something like it), I spent a fair while poking around in my code on the basis that "there's no way `-o` could be broken".

I'll fix that path though, it was fairly unclear if the empty string would do what I wanted.

File: src/librustc_driver/driver.rs (hunk 1 above)
Comment (richo) [inline src/librustc_driver/driver.rs]: Gah, it won't let me comment above the diff. Should the `PathBuf` on line 930 also be the empty string?

Comment (alexcrichton) [inline src/librustc_driver/driver.rs]: Yes that's fine.

**Comment by richo** [comment]
I rebased this to fix the test, but it looks like it'll go out in a rollup anyway.

**Comment by alexcrichton** [comment]
@bors: r+ a837be10c939f38c2b8531e36aea7d2fa57af2df

In general though I would avoid rebasing unless necessary, especially if it's part of a rollup already.

**Comment by Manishearth** [comment]
(This merged, https://github.com/rust-lang/rust/commit/bfcf53f7ad6de8cda2ee4367b2de001fa97d8b0b. The test was fixed in a separate commit.)

**Comment by richo** [comment]
Sure. My reasoning was basically to optimise for not wasting 2 hours of builder time if the rollup failed. Thanks!

**Comment by Manishearth** [comment]
In general if you have to fix something when it's already been included in a rolluo, write it as a separate commit and comment on the rollup so that I can cherry pick. Thanks!

**Pull request closed without merging**